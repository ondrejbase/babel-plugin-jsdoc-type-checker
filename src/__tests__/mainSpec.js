import { transform } from 'babel-core';
import plugin from '../main';

describe('main', () => {
	const ORIGINAL_ENV = process.env;
	const TRANSFORM_OPTIONS = { plugins: [plugin] };

	function setNodeEnv(env) {
		process.env = Object.assign({}, process.env, { NODE_ENV: env });
	}

	beforeEach(() => {
		setNodeEnv('dev');
	});

	afterEach(() => {
		process.env = ORIGINAL_ENV;
	});

	it('should', () => {
		const { code } = transform(
			`
			/**
			 * Class Foo.
			 */
			class Foo {
				/**
				 * Adds up two numbers.
				 *
				 * @param {number} x A first number.
				 * @param {number} y A second number.
				 * @return The sum.
				 */
				addUp(x, y) {
					return x + y;
				}
			}`,
			TRANSFORM_OPTIONS
		);
		expect(code).toMatchSnapshot();
	});
});
