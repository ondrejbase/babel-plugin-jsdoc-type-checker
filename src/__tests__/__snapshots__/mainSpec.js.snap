// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main should 1`] = `
"
/**
 * Class Foo.
 */
class Foo {
	/**
  * Adds up two numbers.
  *
  * @param {number} x A first number.
  * @param {number} y A second number.
  * @return The sum.
  */
	addUp(x, y) {
		return x + y;
	}
}"
`;
